### PR TITLE
[RM Ch05] Log Management Service Reqts #2960

### LINKED ISSUE #2960: [RM Ch05] Log Management Servicve Requirements
This issue is for the specification of the Log Management Service requirements

Umbrella Issue #2785

### PR DESCRIPTION
**Pull request by pgoyal01**
Fixes #2960

### CODE REVIEW

File: doc/ref_model/chapters/chapter05.rst [hunk 1]
```diff
@@ -169,6 +169,39 @@ Tracing                                Jaeger
 
 **Table 5-7a:** Service examples.
 
+Platform Services - Log Management Service (LMS)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The table below specifies a set of requirements for the Log Management Service (LMS).
+
++-------------+-----------------------------------------------------------------------+---------------------------------------+
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter05.rst:177]: Wrap the content to 120 chars please.

File: doc/ref_model/chapters/chapter05.rst [hunk 2]
```diff
@@ -169,6 +169,39 @@ Tracing                                Jaeger
...
+| Reference   | Requirement                                                           | Notes                                 |
++=============+=======================================================================+=======================================+
+| pas.lms.001 | LMS must support log management from multiple, distributed sources    |                                       |
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter05.rst:180]: Is LMS in the glossary, or is it explained before?

File: doc/ref_model/chapters/chapter05.rst [hunk 3]
```diff
@@ -169,6 +169,39 @@ Tracing                                Jaeger
...
+|             | settings, configurations, and processes                               |                                       |
+|             | LMS must support the correlating of entries from multiple logs that   |                                       |
+|             | relate to the same event                                              |                                       |
+|             | LMS must support the correlating of multiple log entries from a       |                                       |
+|             | single source or multiple sources based on logged values (e.g., event |                                       |
+|             | types, timestamps, IP addresses                                       |                                       |
+|             | LMS should support rule-based correlation                             |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+
+**Table 5-7d:** Platform Services - Log Management Service (LMS) Requirements.
```
Comment (CsatariGergely) [inline doc/ref_model/chapters/chapter05.rst:202]: Aham, LMS is expained here, but I think it should be explained at the first occurrence.

**Review by CsatariGergely — CHANGES_REQUESTED** [review]

File: doc/ref_model/chapters/chapter05.rst [hunk 4]
```diff
@@ -169,6 +169,51 @@ Tracing                                Jaeger
...
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.002 | LMS must manage log rotation at configurable time periods             |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.003 | LMS must manage log rotation at configurable log file status (%full)  |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.004 | LMS must manage archival and retention of logs for configurable       |                                       |
+|             | time periods by different log types                                   |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.005 | LMS should ensure log file integrity                                  |                                       |
```
Comment (walterkozlowski) [inline doc/ref_model/chapters/chapter05.rst:189]: pas.lms.005; LMS shoudl ensure log file integrity: what does "integrity" mean in this context?

File: doc/ref_model/chapters/chapter05.rst [hunk 5]
```diff
@@ -169,6 +169,51 @@ Tracing                                Jaeger
...
+| pas.lms.007 | LMS must monitoring the logging status of all log sources             |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.008 | LMS must ensure that each logging host’s clock is synched to a common |                                       |
+|             | time source                                                           |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.009 | LMS must support reconfiguring of logging as needed based on policy   |                                       |
+|             | changes, technology changes, and other factors                        |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.010 | LMS must supppoort the documenting and reporting of anomalies in log  |                                       |
```
Comment (karinesevilla) [inline doc/ref_model/chapters/chapter05.rst:201]: @pgoyal01 typo: suppoort

File: doc/ref_model/chapters/chapter05.rst [hunk 6]
```diff
@@ -169,6 +169,51 @@ Tracing                                Jaeger
...
+|             | settings, configurations, and processes                               |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.011 | LMS must support the correlating of entries from multiple logs that   |                                       |
+|             | relate to the same event                                              |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.012 | LMS must support the correlating of multiple log entries from a       |                                       |
+|             | single source or multiple sources based on logged values (e.g., event |                                       |
+|             | types, timestamps, IP addresses                                       |                                       |
```
Comment (karinesevilla) [inline doc/ref_model/chapters/chapter05.rst:209]: Typo : missing parenthesis at the end

File: doc/ref_model/chapters/chapter05.rst (hunk 4 above)
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter05.rst:189]:
```suggestion
| pas.lms.005 | LMS should ensure log file integrity                                  |  req.sec.mon.005  |
```

File: doc/ref_model/chapters/chapter05.rst [hunk 7]
```diff
@@ -169,6 +169,51 @@ Tracing                                Jaeger
...
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.002 | LMS must manage log rotation at configurable time periods             |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.003 | LMS must manage log rotation at configurable log file status (%full)  |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.004 | LMS must manage archival and retention of logs for configurable       |                                       |
+|             | time periods by different log types                                   |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.005 | LMS should ensure log file integrity                                  |  Covered by req.sec.mon.005  |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter05.rst:189]:
```suggestion
| pas.lms.005 | LMS should ensure log file integrity                                  |  Covered by req.sec.mon.005         |
```

File: doc/ref_model/chapters/chapter05.rst [hunk 8]
```diff
@@ -169,6 +169,51 @@ Tracing                                Jaeger
...
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.002 | LMS must manage log rotation at configurable time periods             |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.003 | LMS must manage log rotation at configurable log file status (%full)  |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.004 | LMS must manage archival and retention of logs for configurable       |                                       |
+|             | time periods by different log types                                   |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.005 | LMS should ensure log file integrity                                  |  Covered by req.sec.mon.005         |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter05.rst:189]:
```suggestion
| pas.lms.005 | LMS should ensure log file integrity                                  |  Covered by req.sec.mon.005           |
```

File: doc/ref_model/chapters/chapter05.rst [hunk 9]
```diff
@@ -169,6 +169,51 @@ Tracing                                Jaeger
...
+| pas.lms.007 | LMS must monitoring the logging status of all log sources             |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.008 | LMS must ensure that each logging host’s clock is synched to a common |                                       |
+|             | time source                                                           |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.009 | LMS must support reconfiguring of logging as needed based on policy   |                                       |
+|             | changes, technology changes, and other factors                        |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.010 | LMS must supppoort the documenting and reporting of anomalies in log  |                                       |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter05.rst:201]:
```suggestion
| pas.lms.010 | LMS must support the documenting and reporting of anomalies in log    |                                       |
```

File: doc/ref_model/chapters/chapter05.rst [hunk 10]
```diff
@@ -169,6 +169,51 @@ Tracing                                Jaeger
...
+| pas.lms.010 | LMS must support the documenting and reporting of anomalies in log    |                                       |
+|             | settings, configurations, and processes                               |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.011 | LMS must support the correlating of entries from multiple logs that   |                                       |
+|             | relate to the same event                                              |                                       |
++-------------+-----------------------------------------------------------------------+---------------------------------------+
+| pas.lms.012 | LMS must support the correlating of multiple log entries from a       |                                       |
+|             | single source or multiple sources based on logged values (e.g., event |                                       |
+|             | types, timestamps, IP addresses                                       |                                       |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter05.rst:209]:
```suggestion
|             | types, timestamps, IP addresses)                                      |                                       |
```

File: doc/ref_model/chapters/chapter05.rst (hunk 3 above)
Comment (pgoyal01) [reply doc/ref_model/chapters/chapter05.rst:214]: @CsatariGergely It is in the title as well as the line preceding the table

**Review by walterkozlowski — APPROVED** [review]

**Review by ulikleber — APPROVED** [review]

**Comment by walterkozlowski** [comment]
@karinesevilla , @pgoyal01 - can this be approved? @karinesevilla, are you satisfied that your comments are addressed?

**Comment by pgoyal01** [comment]
> @karinesevilla , @pgoyal01 - can this be approved? @karinesevilla, are you satisfied that your comments are addressed?

@walterkozlowski @karinesevilla has to re-review.

**Comment by walterkozlowski** [comment]
@karinesevilla, can you please re-review and approve if you can?  Thanks!

**Review by karinesevilla — APPROVED** [review]

**Comment by walterkozlowski** [comment]
@CsatariGergely : can you please check all is addressed and approve if you can, thanks.

**Comment by walterkozlowski** [comment]
@pgoyal01, can you please remove the conflicts, so I can merge?

**Comment by wmk-admin** [comment]
Merged as approved and all comments have been addressed

**Pull request merged**